githubkit/core.py

```diff
@@ -187,7 +198,17 @@ def _get_client_defaults(self):
 
     # create sync client
     def _create_sync_client(self) -> httpx.Client:
-        return httpx.Client(**self._get_client_defaults())
+        if self.config.http_cache:
+            transport = hishel.CacheTransport(
+                httpx.HTTPTransport(), storage=hishel.InMemoryStorage()
+            )
+        else:
+            transport = httpx.HTTPTransport()
```

Comment:
This part may be put into `_get_client_defaults` function to reuse in client creations.

Reply:
That function has no idea whether the client is asynchronous or not. If we move that logic into it, we should also add an argument indicating that it is for an asynchronous client.

Reply:
ok i miss the difference between sync and async transport